addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -6719,6 +6743,15 @@ registry.ImageTools = ImageHandlerOption.extend({
         img.dataset.hoverEffectColor = this._getCSSColorValue(widgetValue || defaultColor);
         await this._applyOptions();
     },
+    /**
+     * Sets the speed of the shape animation.
+     *
+     * @see this.selectClass for params
+     */
+    async setImgShapeAnimationSpeed(previewMode, widgetValue, params) {
+        this.shapeAnimationSpeed = widgetValue;
```

Comment:
This is probably wrong: saving widget values in variable is a great source of bugs for future refactoring and others. Can you explain why you need this variable at all? Can't you:
- apply the new speed on the svg when asked?
- When you need to know the speed value of the current SVG (`_computeWidgetState` for example), read it from the svg directly?

Reply:
> Can you explain why you need this variable at all

In other image options, we always put 'widgetValue' in the image dataset. It's in the 'writeShape' function that we build the SVG with all the information from the dataset. Here, he used a variable instead of the dataset because you asked him to avoid storing information in the DOM. I think the idea of modifying the SVG at this point, in addition to 'writeShape', would complicate things a lot.
It would mean changing the image source for each option separately, rather than doing it all at once.

Reply:
> Here, he used a variable instead of the dataset because you asked him to avoid storing information in the DOM.

Yes, I want to get rid of the dataset but using a variable saved in `this` is worse: it means that if the option is destroyed you code is either broken or inconsistent. With the dataset, it just means the data is duplicated (since the info can be found by reading the SVG).

> I think the idea of modifying the SVG at this point, in addition to 'writeShape', would complicate things a lot.

Is not that already what is done since you call `_applyOptions` just after this line? :thinking: 

> It would mean changing the image source for each option separately, rather than doing it all at once.

Same: is not it already what is done? Each option modifying the shape calls `_applyOptions` immediately.

Reply:
Indeed, `_applyOptions` call `writeShape` later.. But once in `writeShape` we have to know what is the value of each options... Currently it is done thanks the `img dataset` ... I guess you would like to pass the values ​​of the options as parameters of `_applyOptions` ?

Reply:
Probably something like that yes. Maybe we can leave that for later, I don't know, depends how deep a refactoring that would be. I guess it make sense for now to use the dataset to stay consistent with other options then we'll get rid of it all later.